### PR TITLE
Fix offset_end iterator for lists_column_view, which was not correctl…

### PR DESCRIPTION
**Pull request by ttnghia**
Fix the `offset_end` iterator in `lists_column_view`. Since the offset column size is one element larger than the number of column rows, the `offset_end` should be computed as `offset_begin() + size() + 1`. This can also be done by `offset_begin() + offsets().size()`.

This PR blocks https://github.com/rapidsai/cudf/pull/7528, thus it must be merged before that PR.

### CODE REVIEW

File: cpp/include/cudf/lists/lists_column_view.hpp
Comment (jrhemstad) [inline cpp/include/cudf/lists/lists_column_view.hpp:110]: I think this would be a little more intuitive. 
```suggestion
  offset_iterator offsets_end() const noexcept { return offsets_begin() + offsets().size(); }
```

**Review by jrhemstad — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge.

**Comment by kkraus14** [comment]
> @gpucibot merge.

We required two cpp approvals before merging.

**Comment by ttnghia** [comment]
> > @gpucibot merge.
> 
> We require two cpp approvals before merging.

Hah, github tells me "1 approval" 😄

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge.

**Comment by ttnghia** [comment]
@gpucibot merge.

**Review by vuule — APPROVED** [review]

**Comment by kkraus14** [comment]
> Sorry, it's breaking in my other PR. Maybe I miss understood the `breaking` meaning.

Got it. `breaking` and `non-breaking` are referring to downstream consumers of libcudf related to API/ABI/behavior changes that would break their consumption of libcudf.

**Comment by ttnghia** [comment]
@gpucibot merge!!!!!!!!!

Apparently the bot never listens to me.

**Comment by kkraus14** [comment]
Why is this a `breaking` change? This is a bug fix for incorrect results with no API or ABI breaking changes?

**Comment by ttnghia** [comment]
> Why is this a `breaking` change? This is a bug fix for incorrect results with no API or ABI breaking changes?

It should not be any breaking change here. But if I remove the `breaking` label, CI test fails.

**Comment by ttnghia** [comment]
Here we go. Adding `non-breaking` is also a fix. I didn't know that solution.

**Comment by kkraus14** [comment]
There's a label checker that checks that either the `breaking` or `non-breaking` label is set to indicate the behavior. Please ask if there's uncertainty instead of pushing incorrect information.

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by ttnghia** [comment]
> There's a label checker that checks that either the `breaking` or `non-breaking` label is set to indicate the behavior. Please ask if there's uncertainty instead of pushing incorrect information.

Thanks very much, Keith.

**Pull request merged**
